### ISSUE TITLE
perlembed - maintaining a persistent interpreter fails in P5.8

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5927]
Migrated from [rt.perl.org#17230](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=17230) (status was 'resolved')

Searchable as RT17230$

**Comment by p5pRT** [comment]
### From albanez@lex.pl


This is a bug report for perl from albanez@&#8203;lex\.pl\,
generated with the help of perlbug 1\.34 running under perl v5\.8\.0\.




---
if you run 'man perlembed' and do the search for '4th' phrase\, you surely get into 'Maintaining a persistent interpreter' chapter of manual page\. There's an example described in this chapter \- if you try to compile given c program\, following output will apperar&#8203;:

persistent\.c&#8203;: In function \`main'&#8203;:
persistent\.c&#8203;:27&#8203;: \`my\_perl' undeclared \(first use in this function\)
persistent\.c&#8203;:27&#8203;: \(Each undeclared identifier is reported only once
persistent\.c&#8203;:27&#8203;: for each function it appears in\.\)

that bug is happening since P5\.8\, but it appears to not exist in P5\.6




---


---
Flags&#8203;:
&nbsp;    category=library
&nbsp;    severity=high


---
Site configuration information for perl v5\.8\.0&#8203;:

Configured by Debian Project at Fri Sep  6 22&#8203;:29&#8203;:59 UTC 2002\.

Summary of my perl5 \(revision 5\.0 version 8 subversion 0\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=2\.4\.19\, archname=i386\-linux\-thread\-multi
&nbsp;    uname='linux cyberhq 2\.4\.19 \#1 smp sun aug 4 11&#8203;:30&#8203;:45 pdt 2002 i686 unknown unknown gnulinux '
&nbsp;    config\_args='\-Dusethreads \-Duselargefiles \-Dccflags=\-DDEBIAN \-Dcccdlflags=\-fPIC \-Darchname=i386\-linux \-Dprefix=/usr \-Dprivlib=/usr/share/perl/5\.8\.0 \-Darchlib=/usr/lib/perl/5\.8\.0 \-Dvendorprefix=/usr \-Dvendorlib=/usr/share/perl5 \-Dvendorarch=/usr/lib/perl5 \-Dsiteprefix=/usr/local \-Dsitelib=/usr/local/share/perl/5\.8\.0 \-Dsitearch=/usr/local/lib/perl/5\.8\.0 \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/share/man/man3 \-Dman1ext=1 \-Dman3ext=3perl \-Dpager=/usr/bin/sensible\-pager \-Uafs \-Ud\_csh \-Uusesfio \-Uusenm \-Duseshrplib \-Dlibperl=libperl\.so\.5\.8\.0 \-Dd\_dosuid \-des'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
&nbsp;    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fno\-strict\-aliasing \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;    optimize='\-O3'\,
&nbsp;    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fno\-strict\-aliasing \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='2\.95\.4 20011002 \(Debian prerelease\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=4\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib /usr/lib
&nbsp;    libs=\-lgdbm \-ldb \-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    perllibs=\-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    libc=/lib/libc\-2\.2\.5\.so\, so=so\, useshrplib=true\, libperl=libperl\.so\.5\.8\.0
&nbsp;    gnulibc\_version='2\.2\.5'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-rdynamic'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib'

Locally applied patches&#8203;:
&nbsp;    



---
@&#8203;INC for perl v5\.8\.0&#8203;:
&nbsp;    /etc/perl
&nbsp;    /usr/local/lib/perl/5\.8\.0
&nbsp;    /usr/local/share/perl/5\.8\.0
&nbsp;    /usr/lib/perl5
&nbsp;    /usr/share/perl5
&nbsp;    /usr/lib/perl/5\.8\.0
&nbsp;    /usr/share/perl/5\.8\.0
&nbsp;    /usr/local/lib/site\_perl
&nbsp;    \.



---
Environment for perl v5\.8\.0&#8203;:
&nbsp;    HOME=/root
&nbsp;    LANG=C
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/usr/local/sbin&#8203;:/usr/local/bin&#8203;:/usr/sbin&#8203;:/usr/bin&#8203;:/sbin&#8203;:/bin&#8203;:/usr/bin/X11
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @eserte

"albanez@&#8203;lex\.pl \(via RT\)" \<perlbug@&#8203;perl\.org> writes&#8203;:

> \# New Ticket Created by  albanez@&#8203;lex\.pl 
> \# Please include the string&#8203;:  \[perl \#17230\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt2/Ticket/Display\.html?id=17230 >
> 
> 
> 
> This is a bug report for perl from albanez@&#8203;lex\.pl\,
> generated with the help of perlbug 1\.34 running under perl v5\.8\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> 
> if you run 'man perlembed' and do the search for '4th' phrase\, you surely get into 'Maintaining a persistent interpreter' chapter of manual page\. There's an example described in this chapter \- if you try to compile given c program\, following output will apperar&#8203;:
> 
> persistent\.c&#8203;: In function \`main'&#8203;:
> persistent\.c&#8203;:27&#8203;: \`my\_perl' undeclared \(first use in this function\)
> persistent\.c&#8203;:27&#8203;: \(Each undeclared identifier is reported only once
> persistent\.c&#8203;:27&#8203;: for each function it appears in\.\)
> 
> that bug is happening since P5\.8\, but it appears to not exist in P5\.6
> 

Confirmed on FreeBSD\, but only with a threaded perl\. perl without
threads works fine\.

Regards\,
&nbsp;    Slaven

\-\- 
Slaven Rezic \- slaven\.rezic@&#8203;berlin\.de

&nbsp;    tkrevdiff \- graphical display of diffs between revisions \(RCS or CVS\)
&nbsp;    http&#8203;://ptktools\.sourceforge\.net/\#tkrevdiff

**Comment by p5pRT** [comment]
### From @jhi

The examples in perlembed have been corrected and the correct versions
will appear in Perl 5\.8\.1\, whenever that happens\.  If you need a copy of the
new perlembed\, you can get a copy from http&#8203;://www\.iki\.fi/jhi/perlembed\.pod

**Comment by p5pRT** [comment]
@jhi - Status changed from 'new' to 'resolved'
